pkg/library/home/custom_init_test.go

```diff
@@ -0,0 +1,348 @@
+						},
+						InitContainers: []corev1.Container{
+							{
+								Name:  "custom-container",
+								Image: "custom:latest",
+								Args:  []string{"echo 'other init'"},
+							},
+						},
+					},
+				},
+			},
+			expectDefaultInitAdded: true,
+		},
+		{
+			name: "Adds default init when custom init containers list is empty",
+			workspace: &common.DevWorkspaceWithConfig{
+				DevWorkspace: &dw.DevWorkspace{
+					Spec: dw.DevWorkspaceSpec{
+						Template: dw.DevWorkspaceTemplateSpec{
+							DevWorkspaceTemplateSpecContent: dw.DevWorkspaceTemplateSpecContent{
+								Components: []dw.Component{
+									{
+										Name: "test-container",
+										ComponentUnion: dw.ComponentUnion{
+											Container: &dw.ContainerComponent{
+												Container: dw.Container{
+													Image: "test-image:latest",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				Config: &v1alpha1.OperatorConfiguration{
+					Workspace: &v1alpha1.WorkspaceConfig{
+						PersistUserHome: &v1alpha1.PersistentHomeConfig{
+							Enabled:              pointer.Bool(true),
+							DisableInitContainer: pointer.Bool(false),
+						},
+						InitContainers: []corev1.Container{},
+					},
+				},
+			},
+			expectDefaultInitAdded: true,
+		},
+		{
+			name: "Skips default init when DisableInitContainer is true even with custom init",
+			workspace: &common.DevWorkspaceWithConfig{
+				DevWorkspace: &dw.DevWorkspace{
+					Spec: dw.DevWorkspaceSpec{
+						Template: dw.DevWorkspaceTemplateSpec{
+							DevWorkspaceTemplateSpecContent: dw.DevWorkspaceTemplateSpecContent{
+								Components: []dw.Component{
+									{
+										Name: "test-container",
+										ComponentUnion: dw.ComponentUnion{
+											Container: &dw.ContainerComponent{
+												Container: dw.Container{
+													Image: "test-image:latest",
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+				Config: &v1alpha1.OperatorConfiguration{
+					Workspace: &v1alpha1.WorkspaceConfig{
+						PersistUserHome: &v1alpha1.PersistentHomeConfig{
+							Enabled:              pointer.Bool(true),
+							DisableInitContainer: pointer.Bool(true),
+						},
+					},
+				},
+			},
+			expectDefaultInitAdded: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result, err := AddPersistentHomeVolume(tt.workspace)
+			assert.NoError(t, err)
+			assert.NotNil(t, result)
+
+			// Check if default init component was added
+			hasDefaultInit := false
+			for _, component := range result.Components {
+				if component.Name == constants.HomeInitComponentName {
+					hasDefaultInit = true
+					break
+				}
+			}
+
+			if tt.expectDefaultInitAdded {
+				assert.True(t, hasDefaultInit, "Expected default init component to be added")
+			} else {
+				assert.False(t, hasDefaultInit, "Expected default init component NOT to be added")
+			}
```

Comment:
Here we are verifying whether components have default init container or not. We should also verify the initContainer content matches what was configured by user.